### PR TITLE
Bug fix/agency inquiry 503 fix

### PR DESCRIPTION
**Pull request by kvahed**
### Scope & Purpose

*Fixes a bug, which the new testagent revealed in async agency comm's handling of 503 responses from agents.*

- [X] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [ ] Backports required for: *3.7*, *3.6*

#### Related Information

*BTS-356*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/14780/

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

File: arangod/Agency/AsyncAgencyComm.cpp [hunk 1]
```diff
@@ -209,9 +209,10 @@ arangodb::AsyncAgencyComm::FutureResult agencyAsyncInquiry(AsyncAgencyCommManage
                     result.response().header.metaByKey(arangodb::StaticStrings::Location);
                 redirectOrError(man, endpoint, location);
                 return ::agencyAsyncInquiry(man, std::move(meta), std::move(body));
+              } else if (result.statusCode() != fuerte::StatusServiceUnavailable) {
```
Comment (neunhoef) [inline arangod/Agency/AsyncAgencyComm.cpp:212]: Please add a comment to explain the rationale.

**Review by neunhoef — APPROVED** [review]
LGTM, but please add one more comment.

File: arangod/Agency/AsyncAgencyComm.cpp (hunk 1 above)
Comment (kvahed) [reply arangod/Agency/AsyncAgencyComm.cpp:212]: done

**Review by mchacki — CHANGES_REQUESTED** [review]
The production code looks reasonable to me.

However I am missing tests.

This file
tests/AsyncAgencyComm/AsyncAgencyCommTest.cpp
Seems to have the ability to write unit tests for exactly this behavior.
It should tests that if we get 307 that we retry some times (2 or 3 is enough for me) until a good response is returned.

**Comment by neunhoef** [comment]
This is probably obsolete by https://github.com/arangodb/arangodb/pull/13942 but the test is still missing there.

**Comment by neunhoef** [comment]
Obsolete.

**Pull request closed without merging**